core/src/main/java/org/elasticsearch/node/Node.java

```diff
@@ -460,6 +463,7 @@ protected Node(final Environment environment, Collection<Class<? extends Plugin>
                 b.bind(UsageService.class).toInstance(usageService);
                     b.bind(NamedWriteableRegistry.class).toInstance(namedWriteableRegistry);
                     b.bind(MetaDataUpgrader.class).toInstance(metaDataUpgrader);
+                    b.bind(TemplateUpgradeService.class).toInstance(templateUpgradeService);
```

Comment:
Good point! I rewrote the test to get rid of guice there. I don't think we need an api for that. I was just trying to prevent a possible race condition in the test, but I found another way to do that without guice.